From: Simon McVittie <smcv@collabora.com>
Date: Tue, 8 Jan 2019 12:39:46 +0000
Subject: trash test: Don't rely on being able to determine mount points

If we can't find the mount point for target or tmp (as currently
happens on Launchpad autobuilders, and perhaps relatedly, on a
development system that uses btrfs), that's probably not great but is
not really the point of this test.

Author: Simon McVittie <smcv@collabora.com>
Bug-Upstream: https://gitlab.gnome.org/GNOME/glib/issues/1643
Forwarded: https://gitlab.gnome.org/GNOME/glib/merge_requests/577
Origin: https://gitlab.gnome.org/smcv/glib/commit/ad5c289e978178c5d687bff735e9badfe1a6cc09
---
 gio/tests/trash.c | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/gio/tests/trash.c b/gio/tests/trash.c
index b7df936..4bf8fd0 100644
--- a/gio/tests/trash.c
+++ b/gio/tests/trash.c
@@ -121,11 +121,40 @@ test_trash_symlinks (void)
     }
 
   target_mount = g_unix_mount_for (target, NULL);
+
+  if (target_mount == NULL)
+    {
+      gchar *message;
+
+      message = g_strdup_printf ("Unable to determine mount point for %s",
+                                 target);
+      g_test_skip (message);
+      g_free (message);
+      g_free (target);
+      return;
+    }
+
   g_assert_nonnull (target_mount);
   g_test_message ("Target: %s (mount: %s)", target, g_unix_mount_get_mount_path (target_mount));
 
   tmp = g_dir_make_tmp ("test-trashXXXXXX", &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (tmp);
   tmp_mount = g_unix_mount_for (tmp, NULL);
+
+  if (tmp_mount == NULL)
+    {
+      gchar *message;
+
+      message = g_strdup_printf ("Unable to determine mount point for %s", tmp);
+      g_test_skip (message);
+      g_free (message);
+      g_unix_mount_free (target_mount);
+      g_free (target);
+      g_free (tmp);
+      return;
+    }
+
   g_assert_nonnull (tmp_mount);
   g_test_message ("Tmp: %s (mount: %s)", tmp, g_unix_mount_get_mount_path (tmp_mount));
 
